### PR TITLE
Extend PyROS Model Constraint standardization to range Constraints and `ConstraintList` entries

### PR DESCRIPTION
**Pull request by shermanjasonaf**
## Summary/Motivation:
Currently,  `contrib.pyros.util.transform_to_standard_form` does not appropriately address:
- range constraints. In this case, the function removes the `upper` attribute, and recasts the expression to `con.lower - con.body <= 0`.
- `ConstraintList` objects containing at least one entry for which `lower` is not `None`. In this case, only the last standardized entry of the list remains in the model after the function terminates.

## Changes proposed in this PR:
Extend the `transform_to_standard_form` function of `contrib.pyros.util` to appropriately handle:
- range constraints. In particular, if both `lower` and `upper` are identical, the Constraint is recast to the equality `con.body == con.lower`. Otherwise, the constraint is split into two standardized inequalities.
- entries of a `ConstraintList`.
 
Additionally, add testing enhancements.

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by jsiirola — APPROVED** [review]
This looks good.

**Review by blnicho — APPROVED** [review]

File: pyomo/contrib/pyros/util.py [hunk 1]
```diff
@@ -266,19 +266,53 @@ def add_bounds_for_uncertain_parameters(model, config):
...
+
+    Note
+    ----
+    If `a` and `b` are identical and the constraint is not classified as an
+    equality (i.e. the `equality` attribute of the constraint object
+    is `False`), then the constraint is recast to the equality `g(v) == a`.
+    """
+    cons = [con for con in model.component_data_objects(Constraint,
+                                                        descend_into=True,
+                                                        active=True)].copy()
```
Comment (blnicho) [inline pyomo/contrib/pyros/util.py:289]: Why do you need `.copy()` here?

Comment (jsiirola) [reply pyomo/contrib/pyros/util.py:289]: Good catch, @blnicho.  This should only be:
```python
cons = list(model.component_data_objects(Constraint, descend_into=True, active=True))
```

Comment (jsiirola) [reply pyomo/contrib/pyros/util.py]: (fixed with a note why we need to create the intermediate list)

File: pyomo/contrib/pyros/util.py [hunk 2]
```diff
@@ -284,9 +284,11 @@ def transform_to_standard_form(model):
     equality (i.e. the `equality` attribute of the constraint object
     is `False`), then the constraint is recast to the equality `g(v) == a`.
     """
-    cons = [con for con in model.component_data_objects(Constraint,
-                                                        descend_into=True,
-                                                        active=True)].copy()
+    # Note: because we wwill be adding / modifying the number of
```
Comment (blnicho) [inline pyomo/contrib/pyros/util.py:287]: Typo: will

**Review by blnicho — APPROVED** [review]

**Pull request merged**